### PR TITLE
MRC single question decoding pipeline

### PR DESCRIPTION
**Pull request by mihaela-bornea**
This is a pipeline class and notebook for single question MRC decoding.

### CODE REVIEW

File: primeqa/pipelines/extractive_mrc_pipeline.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+from transformers import DataCollatorWithPadding
+from primeqa.mrc.processors.postprocessors.extractive import ExtractivePostProcessor
+from primeqa.mrc.processors.postprocessors.scorers import SupportedSpanScorers
+
+from primeqa.mrc.trainers.mrc import MRCTrainer
+from datasets import Dataset
+import json
+
+class MRCPipeline():
```
Comment (seirasto) [inline primeqa/pipelines/extractive_mrc_pipeline.py:14]: add docstring

**Review by seirasto — APPROVED** [review]
LGTM

File: primeqa/pipelines/extractive_mrc_pipeline.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+
+from primeqa.mrc.trainers.mrc import MRCTrainer
+from datasets import Dataset
+import json
+
+class MRCPipeline():
+    
+    def __init__(self,model_for_qa):
+        task_heads = EXTRACTIVE_HEAD
```
Comment (rongzhangibm) [inline primeqa/pipelines/extractive_mrc_pipeline.py:17]: task_heads should be obtained from input arguments. 'EXTRACTIVE_HEAD' can be used as the default value. Now there are two task_heads ready for use.

File: primeqa/pipelines/extractive_mrc_pipeline.py [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+        config.sep_token_id = tokenizer.convert_tokens_to_ids(tokenizer.sep_token)
+        model = ModelForDownstreamTasks.from_config(
+            config,
+            model_for_qa,
+            task_heads=task_heads,
+            #cache_dir=model_args.cache_dir,
+        )
+        model.set_task_head(next(iter(task_heads)))        
+
+        self.preprocessor = BasePreProcessor(
```
Comment (rongzhangibm) [inline primeqa/pipelines/extractive_mrc_pipeline.py:35]: We should allow user to use different preprocessor and postprocessor.

**Review by rongzhangibm — APPROVED** [review]
LGTM.
I understand that this pipeline is the first version mainly for notebook demo, and will be elaborated in future to allow more flexibilities, e.g. it should allow user to choose different type of task_head, pre/postprocessors, and use different parameter values.

**Pull request merged**